### ISSUE TITLE
NaN loss during training with RTX3090

### ISSUE THREAD

**Issue by urbste** [TRI-ML/KP2D#20]
Does anyone else have this problem? 
I trained the network on a GTX1050ti, RTX2080ti -> no problems.

When I try to train on a new RTX3090 with CUDA 11.1 and either PyTorch 1.8.1 or PyTorch 1.7.1 I get NaNs after a a couple of batches during the training process. 
I tried pip and conda installations nothing seems to work. 

Some other networks I use are not affected during inference. Also training a NeRF model worked without issues. 

Nvidia driver is: 455.32.00

**Comment by RaresAmbrus** [comment]
Hi @urbste - thanks for the interest! I updated the repo from pytorch 1.1 to 1.6 a while ago, and that was a pain as some functions had changed, as well underlying dependencies. I never tried on RTX3090, but I suspect a combination of new hardware and changing dependencies is what's causing the NaNs. 

A couple of things I can think of: are you doing this in docker? Could you maybe try running on RTX3090 but with the docker configuration from the repo, to first debug running on the new card with the old software? The version of opencv you're using also affects performance - I just checked in the docker and we're on `4.4.0`  and `1.19.4` for numpy -- maybe try force-installing those to see if that makes any difference. Let me know how it goes.

**Comment by urbste** [comment]
The problem with PyTorch 1.6.0 is, that it does not come with CUDA 11.x binaries. However without CUDA 11 I will get an PyTorch error: https://discuss.pytorch.org/t/rtx-3090-compatibility/110014
So I would have to build PyTorch from source.

Downgrading the OpenCV version did not help.

**Comment by urbste** [comment]
I am going to close this issue. 
It does not seem that it is directly connected to this repo:
https://github.com/tensorflow/models/issues/9525

Thanks four open sourcing your work btw! :)

**Comment by RaresAmbrus** [comment]
Interesting, thanks for letting me know!
